pandas/core/indexes/interval.py

```diff
@@ -259,6 +259,8 @@ def _simple_new(cls, array, name, closed=None):
         closed : Any
             Ignored.
         """
+        assert isinstance(array, IntervalArray), type(array)
```

Comment:
Not super familiar with `_simple_new` but does this not contradict the base class requiring an `ndarray` or subclass thereof? 

I think passing because `ndarray` right now resolves to `Any`, but if that didn't happen I think mypy would reasonably complain about this

Reply:
if you're suggesting that i remove the annotation on Index._simple_new, sure